client/state/selectors/is-private-site.js

```diff
@@ -12,9 +13,17 @@ import { getRawSite } from 'state/sites/selectors';
  */
 export default function isPrivateSite( state, siteId ) {
 	const site = getRawSite( state, siteId );
-	if ( ! site ) {
-		return null;
+
+	if ( site ) {
+		return site.is_private;
+	}
+
+	const settings = getSiteSettings( state, siteId );
+
+	if ( settings ) {
+		// Site settings returns a numerical value for blog_public.
+		return parseInt( settings.blog_public, 10 ) !== 1;
```

Comment:
I'd prefer to keep the current behavior. I feel like I've seen multiple behaviors while testing this out.